### PR TITLE
Nix speed improvements

### LINKED ISSUE #48: nix: Decrease build time
If we notice that nix building time is too long, we should find a way to reduce it.

Idea:

After successfull build, cache the paths in a file and fetch them from there when changing the PATH,

### PR DESCRIPTION
**Pull request by zuzuleinen**
Closes #48

### CODE REVIEW

File: bobtask/map.go [hunk 1]
```diff
@@ -3,13 +3,11 @@ package bobtask
...
 	"fmt"
-	"path/filepath"
-	"sort"
-
-	"github.com/benchkram/bob/pkg/nix"
-
 	"github.com/benchkram/bob/pkg/multilinecmd"
+	"github.com/benchkram/bob/pkg/nix"
 	"github.com/benchkram/errz"
+	"path/filepath"
```
Comment (Equanox) [inline bobtask/map.go:9]: Was this auto formatted by your IDE?

File: pkg/nix/nix.go [hunk 2]
```diff
@@ -31,22 +32,40 @@ func IsInstalled() bool {
...
 // nixpkgs can be empty which means it will use local nixpkgs channel
 // or a link to desired revision ex. https://github.com/NixOS/nixpkgs/archive/eeefd01d4f630fcbab6588fe3e7fffe0690fbb20.tar.gz
-func BuildDependencies(deps []Dependency) (DependenciesToStorePathMap, error) {
+func BuildDependencies(deps []Dependency) (_ DependenciesToStorePathMap, err error) {
+	defer errz.Recover(&err)
+
+	c, err := NewCacheStore()
+	errz.Fatal(err)
+	defer func() {
+		err = c.Close()
```
Comment (Equanox) [inline pkg/nix/nix.go:41]: Would be ok for me to ignore the error.

Comment (Equanox) [inline pkg/nix/nix.go:41]: Would be ok for me to ignore the error.

File: bob/run.go [hunk 3]
```diff
@@ -48,6 +47,8 @@ func (b *B) Run(ctx context.Context, runName string) (_ ctl.Commander, err error
 	interactiveTasks := []string{runTask.Name()}
 	interactiveTasks = append(interactiveTasks, childInteractiveTasks...)
 
+	//
```
Comment (Equanox) [inline bob/run.go:50]: Remove

File: pkg/nix/cache.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"github.com/benchkram/bob/bob/global"
+	"github.com/benchkram/bob/pkg/filehash"
+	"github.com/benchkram/errz"
+	"os"
```
Comment (Equanox) [inline pkg/nix/cache.go:11]: Put the standard lib packages to the top

File: bobtask/map.go (hunk 1 above)
Comment (zuzuleinen) [reply bobtask/map.go:9]: Yes, my IDE used gofmt import ordering. I switched to use goimports now

File: pkg/nix/nix.go (hunk 2 above)
Comment (zuzuleinen) [reply pkg/nix/nix.go:41]: I thinki here I wanted fatal, not log

**Review by Equanox — APPROVED** [review]

**Pull request merged**